server/proto/merger_test.go

```diff
@@ -0,0 +1,141 @@
+	actual, err := stencilProto.Merge(current, previous)
+	assert.Nil(t, err)
+	actualFDS := &descriptorpb.FileDescriptorSet{}
+	err = proto.Unmarshal(actual, actualFDS)
+	assert.Nil(t, err)
+	actualFiles, err := protodesc.NewFiles(actualFDS)
+	assert.Nil(t, err)
+	assert.Equal(t, expectedFiles.NumFiles(), actualFiles.NumFiles())
+
+	actualFiles.RangeFiles(func(actualFD protoreflect.FileDescriptor) bool {
+		expectedFiles.RangeFiles(func(expectedFD protoreflect.FileDescriptor) bool {
+			if actualFD.Path() == expectedFD.Path() {
+				assert.Equal(t, expectedFD.FullName(), actualFD.FullName())
+				assert.Equal(t, expectedFD.Package(), actualFD.Package())
+				assert.Equal(t, expectedFD.Options(), actualFD.Options())
+				for i := 0; i<expectedFD.Imports().Len(); i++ {
+					missingImport := true
+					for j := 0; j< actualFD.Imports().Len(); j++ {
+						if expectedFD.Imports().Get(i) == actualFD.Imports().Get(j) {
+							missingImport = false
+						}
+					}
+					assert.True(t, missingImport)
+				}
+
+				actualMDs := actualFD.Messages()
+				expectedMDs := expectedFD.Messages()
+				for i := 0; i < expectedMDs.Len(); i++ {
+					expectedMD := expectedMDs.Get(i)
+					actualMD := actualMDs.ByName(expectedMD.Name())
+					assert.NotNil(t, actualMD)
+
+					assertDescriptors(t, expectedMD, actualMD)
+				}
+			}
+			return true
+		})
+		return true
+	})
+}
+
+func assertDescriptors(t *testing.T, expected, actual protoreflect.MessageDescriptor) {
+	if expected.Messages() != nil {
+		expectedNestedMessages := expected.Messages()
+		actualNestedMessages := actual.Messages()
+
+		for i := 0; i < expectedNestedMessages.Len(); i++ {
+			expectedNestedMsg := expectedNestedMessages.Get(i)
+			actualNestedMsg := actualNestedMessages.ByName(expectedNestedMsg.Name())
+			assert.NotNil(t, actualNestedMsg)
+
+			assertDescriptors(t, expectedNestedMsg, actualNestedMsg)
+		}
+	}
+
+	assert.Equal(t, expected.Name(), actual.Name())
+	assert.Equal(t, expected.FullName(), actual.FullName())
+	assert.Equal(t, expected.Options(), actual.Options())
+	expectedFieldDescs := expected.Fields()
+	actualFieldDescs := actual.Fields()
+	assert.Equal(t, expectedFieldDescs.Len(), actualFieldDescs.Len())
+
+	for j := 0; j < expectedFieldDescs.Len(); j++ {
+		expectedFieldDesc := expectedFieldDescs.Get(j)
+		actualFieldDesc := actualFieldDescs.Get(j)
+		assert.Equal(t, expectedFieldDesc.FullName(), actualFieldDesc.FullName())
+		assert.Equal(t, expectedFieldDesc.Number(), actualFieldDesc.Number())
+		assert.Equal(t, expectedFieldDesc.Cardinality(), actualFieldDesc.Cardinality())
+		assert.Equal(t, expectedFieldDesc.Kind(), actualFieldDesc.Kind())
+		assert.Equal(t, expectedFieldDesc.JSONName(), actualFieldDesc.JSONName())
+		assert.Equal(t, expectedFieldDesc.Options(), actualFieldDesc.Options())
+	}
+}
+
+func getTestData(t *testing.T, testNumber int) ([]byte, []byte, []byte) {
+	existingRoot, _ := filepath.Abs(fmt.Sprintf("./testdata/merge/existing/%d", testNumber))
```

Comment:
Can we structure test folder structure as `1/existing, 1/new, 1/expected`, so that it's easy to check test cases?

Reply:
Or let's merge test cases into multiple files with comments on top of the proto message.